### ISSUE TITLE
Augment | TREC-6 DataBunch → Compare with Related Works

### ISSUE THREAD

**Issue by tianjianjiang** [anz9990/nlp_data_aug#14]
- [x] Prepare TREC-6 DataBunch
- [x] Experiment
- [x] Compare with Related Works

**Comment by the-asir** [comment]
@tianjianjiang Please review the latest commit on master and notebook `TREC6`

**Comment by tianjianjiang** [comment]
@anz9990:
I noticed that the naming conventions of the TREC datasets among related papers are confusing. I suppose you’re using TREC-10, and ULMFit paper seems using TREC-8 (they probably call it TREC-6 for the number of classes). I haven’t checked EDA paper.

**Comment by tianjianjiang** [comment]
> @anz9990:
> I noticed that the naming conventions of the TREC datasets among related papers are confusing. I suppose you’re using TREC-10, and ULMFit paper seems using TREC-8 (they probably call it TREC-6 for the number of classes). I haven’t checked EDA paper.

I just double-checked and the situation was:
* ULMFit paper copy-pasted McCann et al. (2017)'s description, which was TREC-8 and allegedly  a bit smaller. However, when I check the actual size of training set showed in your notebook, it appears the same with McCann et al. (2017)'s summary (but different from torchnlp's);
* ULMFit paper also compared with two other papers that claimed they used a TREC-10-based dataset (Li and Roth, 2002), which contains TREC-8 and TREC-9;
* EDA paper uses Li and Roth (2002)'s dataset, too.

In sum, your dataset should be fine.

**Comment by tianjianjiang** [comment]
@anz9990 In case you haven't noticed it, I added a quote of hyperparameters in [README.md](/anz9990/nlp_data_aug/blob/master/README.md) yesterday. If you know where to find hyperparameters for `TREC` (and `AG`), please let me know.

**Comment by the-asir** [comment]
From [paper](https://arxiv.org/pdf/1801.06146.pdf), section 4.1 sub topic Hyperparameters: 
>To this end, if not mentioned otherwise, we use the same set of hyperparameters across tasks,  which we  tune  on  the  IMDb  validation  set

Doesn't this imply the same hyperparameters were used for TREC and AG (except the number of epochs as quoted in the footnote of the same page)?

**Comment by tianjianjiang** [comment]
> From [paper](https://arxiv.org/pdf/1801.06146.pdf), section 4.1 sub topic Hyperparameters:
> 
> > To this end, if not mentioned otherwise, we use the same set of hyperparameters across tasks,  which we  tune  on  the  IMDb  validation  set
> 
> Doesn't this imply the same hyperparameters were used for TREC and AG (except the number of epochs as quoted in the footnote of the same page)?

Only If we are using the original script of the paper. Unfortunately, the script was based on an old version of fastai, and neither the course nor the example ipynb use the same hyper-parameters to the paper.

That being said, I'm trying to simplify all those hyper-parameters like what the paper did: keep batch size and dropouts the same between language modeler and classifier, and then adapt different learning rates.

**Comment by tianjianjiang** [comment]
For TREC to reach the state-of-the-art accuracy, the batch size must be smaller than other bigger datasets.
